clang/lib/CIR/CodeGen/CIRGenCall.cpp

```diff
@@ -395,13 +395,15 @@ RValue CIRGenFunction::buildCall(const CIRGenFunctionInfo &CallInfo,
                "swift NYI");
 
         // We might have to widen integers, but we should never truncate.
-        assert(ArgInfo.getCoerceToType() == V.getType() && "widening NYI");
+        if (ArgInfo.getCoerceToType() != V.getType() &&
+            V.getType().isa<mlir::cir::IntType>())
```

Comment:
What if we have a floating point or some other CIR type? Should this instead be `!V.getType().isa<mlir::cir::VoidTy>()`?

Reply:
@bcardosolopes my understanding is that this clause should only be true when an integer needs to be extended to be passed as a call argument. So it should only be true if the arg is an integer type. Take a look at the [original codegen](https://github.com/llvm/llvm-project/blob/28fb39f16af1003e53008b75c11127b3288742f8/clang/lib/CodeGen/CGCall.cpp#L5210-L5213).